### PR TITLE
Add failing get recursive test

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
Adding a test case for recursive .get that is failing with:

```
    Error: unsupported version
    at CID.get buffer [as buffer] (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipfs-unixfs-engine/node_modules/cids/src/index.js:75:15)
    at Bitswap._getStreamSingle (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipfs-bitswap/src/index.js:139:23)
    at Bitswap.getStream (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipfs-bitswap/src/index.js:121:19)
    at BlockService.getStream (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipfs-block-service/src/index.js:76:28)
    at IPLDResolver.getStream (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipld-resolver/src/index.js:146:15)
    at visitor (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipfs-unixfs-engine/src/exporter/index.js:25:20)
    at /Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-traverse/index.js:62:18
    at /Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-stream/throughs/flatten.js:28:16
    at /Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-stream/throughs/flatten.js:28:16
    at /Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-stream/sources/values.js:21:7
    at nextChunk (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-stream/throughs/flatten.js:20:9)
    at /Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-stream/throughs/flatten.js:41:11
    at drain (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-paramap/index.js:18:11)
    at /Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/pull-paramap/index.js:44:13
    at ipldResolver.get (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipfs-unixfs-engine/src/exporter/dir.js:34:7)
    at pull.collect (/Users/user/projects/ipfs/js-monorepo/packages/js-ipfs/node_modules/ipld-resolver/src/index.js:139:9)
```

### CODE REVIEW

**Comment by diasdavid** [comment]
will look into this one today. @pgte does anything catch your eye there?

**Comment by diasdavid** [comment]
I've discovered that the bug is in unixfs-engine, where instead of having `item.hash` on https://github.com/ipfs/js-ipfs-unixfs-engine/blob/master/src/exporter/index.js#L25, we have item being:

```
{
  content: [Function],
  path: 'QmUhUuiTKkkK8J6JZ9zmj8iNHPuNfGYcszgRumzhHBxEEU/about',
  size: 1677
}
```

This escaped previous testing because we had disabled recursive tests due to js-ipfs-api + go-ipfs browser problems:

https://github.com/ipfs/interface-ipfs-core/blob/master/src/files.js#L371-L431

If those tests are enabled, we can see that now `ipfs files get` also fails on js-ipfs core.

@pgte can you take a look at this one? The new unixfs-engine code is fresh in your mind. I'll try to circle back to this tomorrow if we still haven't found a solution.

**Comment by pgte** [comment]
@VictorBjelkholm as asked by @diasdavid, added test for recursive dir using core. (will also add similar test to js-ipfs-unixfs-engine) next.
Back to you.

File: test/cli/test-files.js [hunk 1]
```diff
@@ -89,5 +89,13 @@ describe('files', () => {
         ].join('\n'))
       })
     })
+
+    it('get recursively', () => {
+      return ipfs('files get QmUhUuiTKkkK8J6JZ9zmj8iNHPuNfGYcszgRumzhHBxEEU').then((out) => {
+        const directory = path.join(process.cwd(), 'QmUhUuiTKkkK8J6JZ9zmj8iNHPuNfGYcszgRumzhHBxEEU')
+        expect(fs.readdirSync(directory).length).to.be.eql(8)
+        // TODO add assertion on content of files
```
Comment (diasdavid) [inline test/cli/test-files.js]: @VictorBjelkholm your turn, tests are passing :) 

(also, make sure to unlink to delete that file, like the other tests do.

**Comment by diasdavid** [comment]
oh, github closed this PR because of the other one being merged. weird @github

**Pull request closed without merging**